src/boot/efi/stub.c

```diff
@@ -180,6 +181,153 @@ static bool use_load_options(
         return true;
 }
 
+static EFI_STATUS get_cmdline_mule(
+                EFI_HANDLE stub_image,
+                EFI_LOADED_IMAGE_PROTOCOL *loaded_image,
+                char16_t **ret_buffer) {
+
+        _cleanup_(file_closep) EFI_FILE *root = NULL, *extra_dir = NULL;
+        size_t dirent_size = 0, n_items = 0, n_allocated = 0;
+        _cleanup_free_ EFI_FILE_INFO *dirent = NULL, *extra_dir_info = NULL;
+        _cleanup_(strv_freep) char16_t **items = NULL;
+        _cleanup_free_ char16_t *buffer = NULL;
+        EFI_STATUS err;
+
+        assert(stub_image);
+        assert(loaded_image);
+        assert(ret_buffer);
+
+        if (!loaded_image->DeviceHandle)
+                goto nothing;
+
+        err = open_volume(loaded_image->DeviceHandle, &root);
+        if (err == EFI_UNSUPPORTED)
+                /* Error will be unsupported if the bootloader doesn't implement the file system protocol on
+                 * its file handles. */
+                goto nothing;
+        if (err != EFI_SUCCESS)
+                return log_error_status(err, "Unable to open root directory: %m");
+
+        err = open_directory(root, u"\\loader\\mules", &extra_dir); // TODO: bikeshedding
```

Comment:
hate the name, just too drug affine

Reply:
maybe just "addon" or "add-on"? 

"conduit", "annex", "augment", "augmentation", "overload", "aux", "supplement"

Reply:
Mules are lovely animals. Also, left the TODO just for the bikeshedding, want to get the functionality in place first, as it requires shim changes.

Reply:
hmm, looking only in a EFI-global dir sounds wrong to me. We should look there too, but the primary focus should really be on a per-uki dir, like we have for creds and sysext

Reply:
sysexts are tied to the image, but to me these addons are generic. Or to say it in another way, the OS vendor provides UKI and sysexts, and the platform vendor provides the cmdline addons, as they are likely tied more to the hardware, than to the UKI. That said doing both in the long term seems fine.